MS-676 Sync config button upgrades

luhmirin-s opened the pull request:
As per updated requirements:
* Added refresh icon (see video below).
* Added a toast notification after both sync workers finish. Since "start" and "end" toasts were shown way too quickly, I decided to change the description text under the preference instead of the "starting" toast (see video below).
* I added a subtitle with the time since the "last update." It is in relative format to keep it consistent with the "last sync" indicator.

https://github.com/user-attachments/assets/48f0b347-b75b-4eb9-9d1a-544d621f4b7e




* As a bonus, I have also added some details about the current configuration to the troubleshooting overview - config ID, config update data (in vulcan) and time since last config sync.

<img src="https://github.com/user-attachments/assets/d8ff08db-2eb6-4d6a-98ba-6f1730f4bdd7" width="480">

jonsimprints commented:
> As per updated requirements:
> 
> * Added refresh icon (see video below).
> * Added a toast notification after both sync workers finish. Since "start" and "end" toasts were shown way too quickly, I decided to change the description text under the preference instead of the "starting" toast (see video below).
> * I added a subtitle with the time since the "last update." It is in relative format to keep it consistent with the "last sync" indicator.
> 
>  Screen_recording_20250113_175317.mp4 
> * As a bonus, I have also added some details about the current configuration to the troubleshooting overview - config ID, config update data (in vulcan) and time since last config sync.
> 
> <img alt="" width="480" src="https://private-user-images.githubusercontent.com/2182889/402606146-d8ff08db-2eb6-4d6a-98ba-6f1730f4bdd7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3OTEyMTksIm5iZiI6MTczNjc5MDkxOSwicGF0aCI6Ii8yMTgyODg5LzQwMjYwNjE0Ni1kOGZmMDhkYi0yZWI2LTRkNmEtOThiYS02ZjE3MzBmNGJkZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTNUMTc1NTE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2I3MDU3ZmQ3M2QwZjE1OGU1MzlhMjc2YTIxM2ZkODA0ZDEwOWQ1ODk4MGExOTdiMTNhODJhNjBmM2Q0ZTkzNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.uXY-L0Y5wAlv9LmpX2d3vv9BbMbt-wtkqsMCyJlsfu4">

This looks brilliabt

meladRaouf commented on feature/dashboard/src/main/java/com/simprints/feature/dashboard/debug/DebugFragment.kt:
```diff
@@ -109,10 +109,6 @@ internal class DebugFragment : Fragment(R.layout.fragment_debug) {
             }
         }
 
```
Do we still need the config button here?

luhmirin-s replied:
Not really. Removed.

luhmirin-s commented:
1. Removed password requirement
2. Removed the random parenthesis from the network

I am unsure that animation is worth the hassle, considering that it would run for a few seconds and is within the performance fragment (I had to implement a custom layout just to add an icon on the right 😄).

luhmirin-s commented:
1. Removed password requirement
2. Removed the random parenthesis from the network

I am unsure that animation is worth the hassle, considering that it would run for a few seconds and is within the performance fragment (I had to implement a custom layout just to add an icon on the right 😄).
